appsec/tests/extension/convert_xml_basic.phpt

```diff
@@ -0,0 +1,100 @@
+--TEST--
+_convert_xml function
+--FILE--
+<?php
+$entity = <<<XML
+<note>
+  <to attr="x">my recipient</to>
+  <from>Jani</from>
+  <tos>
+    <to>John</to>
+    <to>Jane</to>
+  </tos>
+  <heading>Reminder</heading>
+  begin note
+  <p>Don't forget me this weekend!</p>
+  end note
+  <br/>
+  <br myattr="attr value"/>
+</note>
+XML;
+
+$content_type = "application/xml";
+
+//\datadog\appsec\testing\stop_for_debugger();
+$result = datadog\appsec\testing\convert_xml($entity, $content_type);
+if (!$result) {
+print_r(libxml_get_errors());
+}
+
+echo(json_encode($result, JSON_PRETTY_PRINT));
+--EXPECT--
+{
+    "note": {
+        "content": [
+            {
+                "to": {
+                    "content": [
+                        "my recipient"
+                    ],
+                    "attributes": {
+                        "attr": "x"
+                    }
+                }
+            },
+            {
+                "from": {
+                    "content": [
```

Comment:
This layout won't do as the `content` key will shadow the `from` key and it won't allow scanners to properly function.  Something along these lines should preserve the `key:value` relationship

```
"from" [ "Jani", { "attributes": ...}]
```